single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -807,16 +808,14 @@ def stop_charm_services(self):
         self.workload.stop()
 
     @override
-    def restart_charm_services(self):
+    def restart_charm_services(self, force: bool = False):
         """Restarts the charm services with updated config.
 
         If we are running as config-server, we should update both mongod and mongos environments.
         """
-        self.stop_charm_services()
-        self.config_manager.set_environment()
+        self.config_manager.configure_and_restart(force=force)
         if self.state.is_role(MongoDBRoles.CONFIG_SERVER):
-            self.mongos_config_manager.set_environment()
-        self.start_charm_services()
+            self.mongos_config_manager.configure_and_restart(force=force)
```

Comment:
This seems like we are restarting twice, no?

Reply:
No, we restart the mongod and mongos if we are a config-server;
On the config-server, there is a mongod and a mongos running, the mongos running being used only by the charm itself.
We however have to restart both services when needed.